tests/zfs-tests/tests/functional/ctime/ctime.c

```diff
@@ -163,8 +162,9 @@ do_link(const char *pfile)
 	 * Figure out source file directory name, and create
 	 * the link file in the same directory.
 	 */
-	dname = dirname((char *)pfile_copy);
-	(void) snprintf(link_file, BUFSIZ, "%s/%s", dname, "link_file");
+	*strrchr(pfile_copy, '/') = '\0';
```

Comment:
This doesn't need to be changed, does it?

Reply:
Not *really*, but I'd like to add the same thing for basename/dirname as for strtok. Can this executable link to libzfs_core? If so, I'll just link it there. If not, I'll just inline zfs_dirnamelen here.

Reply:
I mean it's not a threaded application, so dirname should be just fine right?

Reply:
It's more about the principle, like strtok. It's fine here, for now. This can change if it grows threads, or uses threads by accident because of some library, so, like strtok, it's easier to forbid tree-wide than it is to validate in perpetuity that all uses are in unthreaded scenarios and allowed to raze the input.

Also, this turns out to be moot, since rewording this bit in terms of zfs_dirnamelen allows it to be half as long, so.